lib/ssl/src/ssl_handshake.erl

```diff
@@ -3538,20 +3525,15 @@ is_supported_sign({Hash, Sign}, 'NULL', _, SignatureSchemes) ->
 %% the certificate of the issuer SHALL apply to the verification of the
 %% signature.
 is_supported_sign({Hash, Sign}, _Param, _, SignatureSchemes) ->
-    Fun = fun (Scheme, Acc) ->
-                  {H0, S0, _} = ssl_cipher:scheme_to_components(Scheme),
+    Fun = fun (Scheme) ->
+                  {H, S0, _} = ssl_cipher:scheme_to_components(Scheme),
                   S1 = case S0 of
                              rsa_pkcs1 -> rsa;
                              S -> S
                          end,
-                  H1 = case H0 of
-                             sha1 -> sha;
-                             H -> H
-                         end,
-                  Acc orelse (Sign  =:= S1 andalso
-                              Hash  =:= H1)
+                  (Sign  =:= S1) andalso (Hash  =:= H)
           end,
-    lists:foldl(Fun, false, SignatureSchemes).
+    lists:any(Fun, SignatureSchemes).
```

Comment:
It seems to me that we have 2 function clauses `is_supported_sign/4` which is doing exactly the same. They are duplicates. The only difference is that the first one is matching for 'NULL' in the second argument while the next one is matching for `_Param` which means don't care. So one of these clauses should be removed

Reply:
Agreed, there is a TODO comment that might sort of explain the two clauses. I will leave the comment and make an issue to address this later, as it is not really part of this fix.